core/src/main/java/org/elasticsearch/index/query/AbstractQueryBuilder.java

```diff
@@ -79,6 +79,16 @@ public final Query toQuery(QueryParseContext parseContext) throws IOException {
         return query;
     }
 
+    /**
+     * Few queries allow you to set the boost and queryName but don't do anything with it. This method allows
+     * to disable boost and queryName for those queries.
+     */
+    //norelease this has to be put in here because of the call on toQuery which would override the boost and queryName
+    //norelease of the wrapped query with defaults
+    protected boolean supportsBoostAndQueryName() {
```

Comment:
I am on the fence on this. We have one query with `boost: 2.0` which holds an inner query with `boost: 3.0`. Let's say that the final lucene query will be the lucene representation of the inner query. What about the final boost, should it be 2.0 or 3.0? Right now the boost set on the main query overrides the inner boost. I am not sure if this is the right behaviour, which would simply require to adjust the doAssertLuceneQuery here to take this behaviour into account. But `supportsBoostAnQueryName` should not be moved to prod code. I would go back to having it in test and I would adjust the doAssertLuceneQuery here to work properly without needing to disable anything. Then we can open an issue and discuss what the final boost should be.